docs/sdk/src/reference_docs/umbrella_crate.rs

```diff
@@ -0,0 +1,89 @@
+//! - `with-tracing`: Enable tracing support.
+//! - `try-runtime`, `runtime-benchmarks` and `std`: These follow the standard conventions.
+//! - `runtime`: As described above, enable all `no-std` crates.
+//! - `node`: As described above, enable all `std` crates.
+//! - There does *not* exist a dedicated docs feature. To generate docs, enable the `runtime` and
+//!   `node` feature. For docs.rs the manifest contains specific configuration to make it show up
+//!   all re-exports.
+//!
+//! There is a specific `zepter` check in place to ensure that the features of the umbrella are
+//! correctly configured. This check is run in CI and locally when running `zepter`.
+//!
+//! ## Generation
+//!
+//! The umbrella crate needs to be updated every time when a new crate is added or removed from the
+//! workspace. It is checked in CI by calling its generation script. The generation script is
+//! located in `./scripts/generate-umbrella.py` and needs dependency `cargo_workspace`.
+//!
+//! Example: `python3 scripts/generate-umbrella.py --sdk . --version 1.9.0`
+//!
+//! ## Usage
+//!
+//! > Note: You can see a live example in the `staging-node-cli` and `kitchensink-runtime` crates.
+//!
+//! The umbrella crate can be added to your runtime crate like this:
+//!
+//! `polkadot-sdk = { path = "../../../../umbrella", features = ["runtime"], default-features =
+//! false}`
+//!
+//! or for a node:
+//!
+//! `polkadot-sdk = { path = "../../../../umbrella", features = ["node"], default-features = false
+//! }`
+//!
+//! In the code, it is then possible to bring all dependencies into scope via:
+//!
+//! `use polkadot_sdk::*;`
+//!
+//! ### Known Issues
+//!
+//! The only known issue so far is the fact that the `use` statement brings the dependencies only
+//! into the outer module scope - not the global crate scope. For example, the following code would
+//! need to be adjusted:
+//!
+//! ```rust
+//! use polkadot_sdk::*;
+//!
+//! mod foo {
+//!    // This does sadly not compile:
+//!    frame_support::parameter_types! { }
+//!
+//!    // Instead, we need to do this (or add an equivalent `use` statement):
+//!    polkadot_sdk::frame_support::parameter_types! { }
+//! }
+//! ```
+//!
+//! Apart from this, no issues are known. There could be some bugs with how macros locate their own
+//! re-exports. Please compile issues that arise from using this crate.
+//!
+//! ## Dependencies
+//!
+//! The umbrella crate re-exports all published crates, with a few exceptions:
+//! - Runtime crates like `rococo-runtime` etc are not exported. This otherwise leads to very weird
+//!   compile errors and should not be needed anyway.
+//! - Example and fuzzing crates are not exported. This is currently detected by checking the name
+//!   of the crate for these magic words. In the future, it will utilize custom metadata, as it is
+//!   done in the `rococo-runtime` crate.
+//! - The umbrella crate itself. Should be obvious :)
```

Comment:
@gupnik you can look into the experience of developing a pallet with this vs. using the frame crate, and form an opinion on whether we still need the frame crate or not.

I personally think we can benefit from both. frame crate still helps in aggregating a bunch of commonly used stuff like traits, arithmetic, macros, runtime apis and such under well organized preludes and this is good.